### ISSUE TITLE
Support testgrid annotations in other repos

### ISSUE THREAD

**Issue by howardjohn** [kubernetes/test-infra#13455]
**What would you like to be added**:
I would like to be able to use testgrid annotation in job config that is outside of k/test-infra. See https://github.com/istio/test-infra/pull/1475

**Why is this needed**:
Keeping the testgrid in sync manually is virtually impossible. In order to get testgrid working for us, we will need a process that will automatically open PRs against this repo and keep the testgrid in sync, which is a bit of a pain.

Alternatively, we would be interesting in our own deployment of testgrid when it is open sourced, which would probably solve this problem

**Comment by chases2** [comment]
@fejta 
/assign

Our proposed strategy to deal with this is the following:
- Create a tool that allows prow annotations to be converted to a YAML file that could be placed in a Testgrid Config folder (#13541) via a "bump" script. In this way, external instances of Prow can control and output their testgrid configurations.
- Eventually, once Testgrids can be deployed OS in parallel to an existing prow deployment, the bump script can be removed and replaced with Testgrid's tooling for reading prow annotations.

**Comment by chases2** [comment]
Now that there is [documentation](https://github.com/kubernetes/test-infra/blob/master/testgrid/cmd/configurator/README.md) showing this use case for Configurator, and Istio has used it for this purpose (#13699), I'm closing this issue for now.

Further development will come as one of the general use cases for open-sourcing Testgrid (#10409), and should be tracked there or in a separate issue.

/close
